supdup.c

```diff
@@ -164,7 +165,11 @@ struct	sgttyb ottyb;
 unsigned int sup_ispeed = 9600, sup_ospeed = 9600;
 #else
 struct termios otio;
+# ifndef __HAIKU__
 speed_t sup_ispeed = 9600, sup_ospeed = 9600;
+# else
+unsigned int sup_ispeed = 9600, sup_ospeed = 9600;
+# endif
```

Comment:
What about making a typedef for speed_t if and only if __HAIKU__?

Reply:
I think that's a good suggestion.

Reply:
I'm not familiar with speed_t but on Haiku, it is a signed char or something like that, so it gets set to 128, which doesn't seem to be what is intended.

@ams @larsbrinkhoff is there any problem using unsigned int everywhere?  I can also do a typedef or definition of course.

Reply:
I'll have do a define, seems speed_t is a basic type on Haiku

Reply:
Oh, so there is a speet_t on Haiku, but it is signed char instead of unsigned int? That seems annoying.

Reply:
Yes. :(